lib/src/components/NxToggle/__tests__/NxToggle.test.tsx

```diff
@@ -5,116 +5,97 @@
-        inputId: 'toggle-id',
-        isChecked: false,
-        onChange: () => {},
-        disabled: undefined,
-        children: 'Enables whales'
-      },
-      getShallow = getShallowComponent<Props>(NxToggle, simpleProps);
+    inputId: 'toggle-id',
+    isChecked: false,
+    onChange: () => {},
+    disabled: undefined,
+    children: 'Enables whales'
+  };
 
-  it('renders a <label> containing a toggle <input> and toggle control', function() {
-    const shallowRender = getShallow();
+  it('renders a <label> with an <input>', function() {
+    render(<NxToggle {...simpleProps}></NxToggle>);
 
-    expect(shallowRender).toMatchSelector('label.nx-toggle');
+    const checkbox = screen.getByRole<HTMLInputElement>('switch', { name: 'Enables whales' });
 
-    expect(shallowRender.find('input')).toHaveProp('type', 'checkbox');
-    expect(shallowRender.find('input')).toHaveProp('id', 'toggle-id');
-    expect(shallowRender.find('input')).toHaveClassName('nx-toggle__input');
+    expect(checkbox.tagName).toEqual('INPUT');
 
-    expect(shallowRender.find('.nx-toggle__control')).toExist();
-  });
+    const label = checkbox.labels?.[0];
 
-  it('adds classes specified with the className prop', function() {
-    const component = getShallow({ className: 'foo' });
-
-    expect(component).toHaveClassName('foo');
-    expect(component).toHaveClassName('nx-toggle');
+    expect(label?.textContent).toEqual('Enables whales');
   });
 
-  it('adds the nx-toggle--disabled class if disabled is set', function() {
-    expect(getShallow()).not.toHaveClassName('nx-toggle--disabled');
-    expect(getShallow({ disabled: true })).toHaveClassName('nx-toggle--disabled');
-  });
+  it('renders an input element with role switch with the correct attributes and classname', function() {
+    render(<NxToggle {...simpleProps}></NxToggle>);
 
-  it('adds the tm-checked class if isChecked is true, and the tm-unchecked class if it is false', function() {
-    expect(getShallow()).toHaveClassName('tm-unchecked');
-    expect(getShallow()).not.toHaveClassName('tm-checked');
+    const checkbox = screen.getByRole('switch');
 
-    expect(getShallow({ isChecked: true })).not.toHaveClassName('tm-unchecked');
-    expect(getShallow({ isChecked: true })).toHaveClassName('tm-checked');
+    expect(checkbox).toHaveAttribute('type', 'checkbox');
+    expect(checkbox).toHaveAttribute('id', 'toggle-id');
+    expect(checkbox).toHaveClass('nx-toggle__input');
   });
 
-  it('renders children nodes within an nx-toggle__content <span>', function() {
-    const render = shallow(
-      <NxToggle { ...simpleProps }>
-        <div className="bar"></div>
-      </NxToggle>
-    );
+  it('adds classes specified with the className prop', function() {
+    const { container } = render(<NxToggle {...simpleProps} className="foo"></NxToggle>);
 
-    expect(render).toContainMatchingElement('label .nx-toggle__content .bar');
-  });
+    const label = container.querySelector('label');
```

Comment:
When we're trying to get _the_ top level element that the component renders, is it inappropriate to do something like `container.children[0]`?  I feel like these tests as-written don't do a good job of asserting that the label is the top-level element, which when it comes to testing that custom classes get added, is probably important.

Reply:
> When we're trying to get _the_ top level element that the component renders, is it inappropriate to do something like `container.children[0]`? I feel like these tests as-written don't do a good job of asserting that the label is the top-level element, which when it comes to testing that custom classes get added, is probably important.

That's a good point: I updated it here: 70a802873cd0a388c381c0dab40dd72929eba491